src/plugins/data_source/server/plugin.ts

```diff
@@ -45,6 +53,31 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
       credentialSavedObjectsClientWrapper.wrapperFactory
     );
 
+    this.dataSourceService = new DataSourceService(this.logger, config);
```

Comment:
let's follow the practice and put object creation into plugin constructor if possible

Reply:
agreed, I'll move it